### ISSUE TITLE
DB/Quest: fix availability for quests James Hyal and They Call Him Smiling Jim

### ISSUE THREAD

**Issue by Alastro-Kraken** [TrinityCore/TrinityCore#27725]
### Description

Those two quests interact in a strange way with each other, on Classic TBC retail:

- You can accept [James Hyal (Part 1)](https://tbc.wowhead.com/quest=1301/james-hyal) even if you have accepted [They Call Him Smiling Jim](https://tbc.wowhead.com/quest=1282/they-call-him-smiling-jim).
- You CANNOT accept [James Hyal (Part 1)](https://tbc.wowhead.com/quest=1301/james-hyal) if you already turned in [They Call Him Smiling Jim](https://tbc.wowhead.com/quest=1282/they-call-him-smiling-jim).
- The same goes in the opposite way - you can accept [They Call Him Smiling Jim](https://tbc.wowhead.com/quest=1282/they-call-him-smiling-jim) even if you already accepted [James Hyal (Part 1)](https://tbc.wowhead.com/quest=1301/james-hyal), but NOT if [James Hyal (Part 1)](https://tbc.wowhead.com/quest=1301/james-hyal) is rewarded.
- [James Hyal (Part 2)](https://tbc.wowhead.com/quest=1302/james-hyal) is unrelated to either quest's completion (aside from having [James Hyal (Part 1)](https://tbc.wowhead.com/quest=1301/james-hyal) as breadcrumb), you can do it even after completing the whole Shady Inn quest chain.

Basically, as [this Wowhead comment](https://tbc.wowhead.com/quest=1282/they-call-him-smiling-jim#comments:id=98307:reply=17776) says, you can do all three quests if you accept [James Hyal (Part 1)](https://tbc.wowhead.com/quest=1301/james-hyal) and [They Call Him Smiling Jim](https://tbc.wowhead.com/quest=1282/they-call-him-smiling-jim) at the same time before turning in either, but not if you accept one and turn it in without accepting the other.

This is the only case I know of where you can turn in a quest even if you wouldn't be able to re-accept it if you abandon it. But hey, it's Blizzlike™

EDIT: did some more testing, none of the quests above are actually required to accept [Inspecting the Ruins](https://tbc.wowhead.com/quest=11123/inspecting-the-ruins), in short they're all optional breadcrumbs. Updated fix to remove requirements that were set on DB. Only [They Call Him Smiling Jim](https://tbc.wowhead.com/quest=1282/they-call-him-smiling-jim) is a direct breadcrumb (even on Wowhead they're chained), the other two James Hyal quests are not. I really can't understand what Blizzard was thinking, but I spent two hours creating characters and checking all combinations I could think of on retail, so the fix is final.

### SQL fix

```sql
UPDATE `quest_template_addon` SET `NextQuestID`=0, `ExclusiveGroup`=0 WHERE `ID` IN (1282, 1302);
UPDATE `quest_template_addon` SET `BreadcrumbForQuestId`=11123 WHERE `ID`=1282;
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=19 AND `SourceEntry` IN (1282, 1301);
INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
(19, 0, 1282, 0, 0, 8, 0, 1301, 0, 0, 1, 0, 0, "", "Quest They Call Him Smiling Jim requires quest James Hyal (Part 1) NOT rewarded"),
(19, 0, 1301, 0, 0, 8, 0, 1282, 0, 0, 1, 0, 0, "", "Quest James Hyal (Part 1) requires quest They Call Him Smiling Jim NOT rewarded");

```

### Branch

3.3.5

### TC rev. hash/commit

b8059ec6e755a391bcbf951de81e205ede3cf553

**Comment by TQPS** [comment]
That sounds right. I ran into this a while ago. I completed "James Hyal" in Stormwind and Wetlands but when I went to Theramore I couldn't see "They call him Smiling Jim". Found out if you complete 1 first, you cant do the other.

<details>

![image](https://user-images.githubusercontent.com/66294837/152600034-dac6803f-a594-4211-a348-ab3f84d8856c.png)

</details>

As annoying as it is, it's working as intended.

I believe there's 2 other quests in Darkshore that work the same way. I cant complete them because I've completed the alternate path. Blizzlike - working as intended.
